rotkehlchen/tasks/server.py

```diff
@@ -0,0 +1,168 @@
+
+import gevent
+import zmq.green as zmq
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.args import app_args
+from rotkehlchen.db.settings import ModifiableDBSettings
+from rotkehlchen.errors.misc import GreenletKilledError
+from rotkehlchen.premium.premium import PremiumCredentials
+from rotkehlchen.protobufs import all_common_pb2, bg_worker_pb2
+from rotkehlchen.protobufs.serialization import (
+    PROCESS_CONNECTION_ADDRESS,
+    PROCESS_ERROR_TO_ENUM,
+    from_typed_data,
+)
+from rotkehlchen.rotkehlchen import Rotkehlchen
+from rotkehlchen.types import ChecksumEvmAddress
+
+UnderlyingCursor: TypeAlias = sqlite3.Cursor | sqlcipher.Cursor  # pylint: disable=no-member
+UnderlyingConnection: TypeAlias = sqlite3.Connection | sqlcipher.Connection  # pylint: disable=no-member
+SchedulerMethod = bg_worker_pb2.SchedulerMethod  # type: ignore[attr-defined]  # pylint: disable=no-member
+
+DEFAULT_BG_WORKER_PORT: Final = 5556
+
+
+class TaskManagerServer:
+    """Server that listens for ZMQ messages and runs background tasks.
+
+    Listens on a ZMQ REP socket and runs background tasks based on the start/stop messages
+    it receives. The messages are of type SchedulerMethod which can be either 'SCHEDULE' or 'CLEAR'
+    The server will respond with an empty response, after succesfully processing the message.
+    """
+
+    def __init__(self, args: Namespace) -> None:
+        self.zmq_connection = zmq.Context().socket(zmq.REP)
+        self.zmq_connection.bind(f'{PROCESS_CONNECTION_ADDRESS}:{args.bg_worker_port}')
+        # a flag to gracefully shut down the server and
+        # to avoid sending back error from the server while terminating.
+        self.shutting_down = False
+        self.rotkehlchen = Rotkehlchen(args)
+        self.tasks_greenlet: gevent.Greenlet | None = None
+
+    def _unlock(self, call_data: all_common_pb2.UnlockCall) -> None:  # type: ignore[name-defined]  # pylint: disable=no-member
+        """Unlock the rotkehlchen instance and initialize the task manager."""
+        self.rotkehlchen.unlock_user(
+            user=call_data.username,
+            password=call_data.password,
+            create_new=False,  # it should already be created from the api backend process
+            sync_approval=all_common_pb2.SyncApproval.Name(call_data.sync_approval).lower(),  # type: ignore[attr-defined]  # pylint: disable=no-member
+            premium_credentials=PremiumCredentials(
+                given_api_key=call_data.premium_credentials.api_key,
+                given_api_secret=call_data.premium_credentials.api_secret,
+            ) if call_data.HasField('premium_credentials') else None,
+            resume_from_backup=call_data.resume_from_backup,
+            initial_settings=ModifiableDBSettings.deserialize({
+                key: from_typed_data(data)
+                for key, data in call_data.initial_settings.items()
+            }),
+            sync_database=call_data.sync_database,
```

Comment:
Do you need to send all those fields? Can't those be read form the db?

Reply:
I expect that if you change the premium credentials you would need some kind of refresh